src/translib/transformer/xfmr_ztp.go

```diff
@@ -0,0 +1,339 @@
+package transformer
+
+import (
+    "strconv"
+    "reflect"
+    "translib/db"
+    //"strings"
+    "github.com/openconfig/ygot/ygot"
+    log "github.com/golang/glog"
+    "translib/ocbinds"
+    "encoding/json"
+    "fmt"
+)
+/*App specific constants */
+const (
+    ZTP_STATUS_ADMIN_MODE            = "admin_mode"
+    ZTP_STATUS_SERVICE               = "service"
+    ZTP_STATUS_STATUS                = "status"
+    ZTP_STATUS_SOURCE                = "source"
+    ZTP_STATUS_RUNTIME               = "runtime"
+    ZTP_STATUS_TIMESTAMP             = "timestamp"
+    ZTP_STATUS_JSON_VERSION          = "json_version"
+    ZTP_STATUS_ACTIVITY_STRING       = "activity_string"
+    ZTP_CONFIG_SECTION_LIST          = "config_section_list"
+    ZTP_CONFIG_SECTION_STATUS        = "cfg_status"
+    ZTP_CONFIG_SECTION_NAME          = "cfg_sectionname"
+    ZTP_CONFIG_SECTION_RUNTIME       = "cfg_runtime"
+    ZTP_CONFIG_SECTION_TIMESTAMP     = "cfg_timestamp"
+    ZTP_CONFIG_SECTION_EXITCODE      = "cfg_exitcode"
+    ZTP_CONFIG_SECTION_IGNORE_RESULT = "cfg_ignoreresult"
+    ZTP_STATUS_ERROR                 = "error"
+)
+
+/* App specific type definitions */
+
+type ztpStatusCache struct {
+    ztpStatusMap map[string]string
+    ztpConfigSectionMap map[string]map[string]string
+}
+
+/* App specific utilities */
+
+/* Initialise ZTP cache Data structure */
+
+func ztpCacheInit(statusCache *ztpStatusCache) {
+    statusCache.ztpStatusMap = make(map[string]string)
+    statusCache.ztpConfigSectionMap= make(map[string]map[string]string)
+}
+
+/* Get ygot root object */
+
+func getZtpRoot (s *ygot.GoStruct) (*ocbinds.OpenconfigZtp_Ztp) {
+    deviceObj := (*s).(*ocbinds.Device)
+    return deviceObj.Ztp
+}
+
+/* Wrapper to call host service to perform ZTP operations */
+
+func ztpAction(action string) (string, error) {
+	var output string
+	// result.Body is of type []interface{}, since any data may be returned by
+	// the host server. The application is responsible for performing
+	// type assertions to get the correct data.
+	result := hostQuery("ztp." + action)
+	if result.Err != nil {
+		return output, result.Err
+	}
+	if ((action == "status") || (action == "getcfg")) {
+		// ztp.status returns an exit code and the stdout of the command
+		// We only care about the stdout (which is at [1] in the slice)
+		output, _ = result.Body[1].(string)
```

Comment:
Use `result.Body[0].(string)`, as described in my comment above - we don't need to return the unnecessary "code"